src/event/writer.rs

```diff
@@ -161,9 +172,43 @@ impl EventWriter {
                 .expect("send error");
             rx_error
         } else {
+            self.clear_initial_complete_events();
+            self.event_handles.push_back(rx_flush);
             rx
         }
     }
+
+    /// Flush data.
+    ///
+    /// It will wait until all pending appends have acknowledgment.
+    /// ```
+    pub async fn flush(&mut self) -> Result<(), Error> {
+        self.clear_initial_complete_events();
+        while let Some(receiver) = self.event_handles.pop_front() {
+            let recv = receiver.await.map_err(|e| Error::InternalFailure {
+                msg: format!("oneshot error {:?}", e),
+            })?;
+
+            recv?;
+        }
+        Ok(())
+    }
+
+    /// Clear initial completed events from flush queue.
+    fn clear_initial_complete_events(&mut self) {
+        while let Some(mut receiver) = self.event_handles.pop_front() {
+            let try_recv = receiver.try_recv();
+
+            match try_recv {
+                Err(TryRecvError::Empty) => {
+                    self.event_handles.push_front(receiver);
+                    break;
+                }
+                Err(TryRecvError::Closed) => {}
+                _ => {}
```

Comment:
Please make this:
```
Ok(_) => {}
```
That way if a new error value is ever added to `try_receive` we'll get a compile error rather than being unaware of it and simply skipping over the error at runtime.